universe/NamedValueRefManager.cpp

```diff
@@ -0,0 +1,206 @@
+#include "ValueRefs.h"
+
+#include <functional>
+#include <iomanip>
+#include <iterator>
+#include <mutex>
+#include <unordered_map>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/log/expressions.hpp>
+#include <boost/range/adaptor/filtered.hpp>
+#include <boost/range/adaptor/map.hpp>
+#include <boost/range/numeric.hpp>
+#include "Enums.h"
+#include "Field.h"
+#include "Fighter.h"
+#include "Fleet.h"
+#include "Pathfinder.h"
+#include "Planet.h"
+#include "ShipDesign.h"
+#include "ShipHull.h"
+#include "ShipPart.h"
+#include "Ship.h"
+#include "Species.h"
+#include "System.h"
+#include "Tech.h"
+#include "UniverseObjectVisitors.h"
+#include "UniverseObject.h"
+#include "Universe.h"
+#include "../Empire/EmpireManager.h"
+#include "../Empire/Empire.h"
+#include "../Empire/Supply.h"
+#include "../util/GameRules.h"
+#include "../util/Logger.h"
+#include "../util/MultiplayerCommon.h"
+#include "../util/Random.h"
+
+namespace expr = boost::log::expressions;
+
+NamedValueRefManager* NamedValueRefManager::s_instance = nullptr;
+
+NamedValueRefManager::NamedValueRefManager() {
+    if (s_instance)
+        throw std::runtime_error("Attempted to create more than one NamedValueRefManager.");
+
+    // Only update the global pointer on sucessful construction.
+    ErrorLogger() << "NamedValueRefManager::NameValueRefManager constructs singleton " << this;
+    s_instance = this;
+}
+
+NamedValueRefManager::~NamedValueRefManager() {
+    ErrorLogger() << "NamedValueRefManager::~NameValueRefManager destruct " << this ; // FIXME
```

Comment:
error?

Reply:
i did not yet check what is necessary to do in the destructor